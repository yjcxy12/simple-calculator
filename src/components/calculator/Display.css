.calculator-display {
  background-color: white;
  border-bottom: 1px solid rgba(256, 256, 256, 0.7);
  box-sizing: border-box;
  color: #565151;
  font-size: 42px;
  font-weight: 200;
  height: 10%;
  line-height: 42px;
  min-height:73px;
  padding: 15px;
  text-align: right;
  width: 100%;
}
