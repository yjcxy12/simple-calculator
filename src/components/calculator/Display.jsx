import React from 'react';
import './Display.css';

export const Display = ({ value = '0' }) => (
  <div data-testid="display" className="calculator-display">
    {value}
  </div>
);
