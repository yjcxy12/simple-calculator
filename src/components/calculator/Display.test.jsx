import React from 'react';
import renderer from 'react-test-renderer';
import { Display } from './Display';

describe('snapshots', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Display value="23.12" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if no value is provided', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
