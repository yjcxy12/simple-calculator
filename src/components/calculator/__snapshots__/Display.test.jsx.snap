// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders correctly 1`] = `
<div
  className="calculator-display"
  data-testid="display"
>
  23.12
</div>
`;

exports[`snapshots renders correctly if no value is provided 1`] = `
<div
  className="calculator-display"
  data-testid="display"
>
  0
</div>
`;
